torch_xla/csrc/ops/scalar.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "ir.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+// Differently from Constant, this is a scalar value broadcasted to a shape.
+// Even though a Constant could have been used, for simple scalars broadcasted
+// to big shapes, the Constant leads to big literals expanded within the XLA
+// graph.
+class Scalar : public Node {
```

Comment:
Since this has just one purpose, I think `BroadcastScalar` would be a better name.

Reply:
IMHO "broadcasting" is an consequence of the kind of shape (we can even build this with the simple XLA type), so "scalar" might still be a good pick (and shorter).